manager/state/raft/raft.go

```diff
@@ -354,7 +368,23 @@ func (n *Node) done() {
 // Before running the main loop, it first starts the raft node based on saved
 // cluster state. If no saved state exists, it starts a single-node cluster.
 func (n *Node) Run(ctx context.Context) error {
-	defer n.done()
+	ctx, cancel := n.WithContext(ctx)
```

Comment:
Idea that may or may not make sense... what if instead of having a `Stop` call, we let the caller cancel the context passed to `Run` to stop raft? Then we don't need to mess with the context.

We'd need some kind of `Done` method so the manager could wait for raft to stop after canceling the context.